File: delsig/plotSpectrum.m
function plotSpectrum(X,fin,fmt,nbin,n)
% plotSpectrum(X,fin,fmt='-',nbin=8,n=3) Plot a smoothed spectrum
if nargin<5
    n = 3;
    if nargin<4
        nbin = 8;
        if nargin<3
            fmt = '-';
        end
    end
end
i = fmt=='.' | ( fmt>='0' & fmt <='9' );
if any(i)
    i = find(~i); i = i(end); % i = index of last character before the numeric suffix
    lw = str2num(fmt(i+1:end));
    fmt = fmt(1:i);
else
    lw = 0.5;
end
[f p] = logsmooth(X,fin,nbin,n);
semilogx(f,p,fmt,'Linewidth',lw);

File: pydelsigma/_plotSpectrum.py
# -*- coding: utf-8 -*-
# _plotSpectrum.py
# Module providing the plotSpectrum function
# Copyright 2013 Giuseppe Venturini
# This file is part of python-deltasigma.
#
# python-deltasigma is a 1:1 Python replacement of Richard Schreier's 
# MATLAB delta sigma toolbox (aka "delsigma"), upon which it is heavily based.
# The delta sigma toolbox is (c) 2009, Richard Schreier.
#
# python-deltasigma is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# LICENSE file for the licensing terms.

"""Module providing the plotSpectrum() function
"""

import pylab as plt

from ._logsmooth import logsmooth

def plotSpectrum(X, fin, fmt='-'):
    """Plot a smoothed spectrum on the current figure.
    """
    f, p = logsmooth(X, fin)
    plt.semilogx(f, p, fmt)
